interview-questions.md

```diff
@@ -6,53 +6,106 @@ First, without external resources. Challenge yourself to answer from memory.
+  pass properties to children components just like we pass in arguments to functions. If we think of a save point in a video game like saving the "state"
+  of the game, then we can look at transferring a weapon with different "properties" from one character to another as passing props from one component to the next.
 
+****  Reference ****
+https://dmitripavlutin.com/react-props/
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: 
+  A DOM Event is a user or browser initiated action which causes a code block of logic to execute.For example, adding an onClick to a button tag will
+  trigger an action to occur when the user clicks the button. This could be anything from changing the background color of the webpage or adding 
+  text from an input field to an unordered list.
 
   Researched answer:
+  In the example above, adding an onClick to a button tag would be referred to as adding an event handler. The same output can be accomplished by adding 
+  an event listener to an object. For example 
+  button.addEventListener("click", () => {
+  ** Action to be executed **
+  });
+  Just like the name implies, the second way, "listens" for an event to occur then executes the code. It is important to note that multiple event listeners
+  are allowed while only one event handler for any given even is permittable.
 
-
+****  Reference ****
+https://medium.com/geekculture/event-handlers-vs-event-listeners-in-javascript-b4086b8040b0
 
 3. What is object-oriented programming? How is it different than functional programming?
 
   Your answer:
+  I believe object-oriented programming is a set of guidelines many developers use to ensure their code remains consistent
+  throughout the entire project. OOP follows a set rubric as opposed to functional programming where there is no set standard
+  as long as the end goal of a working product is achieved.
 
   Researched answer:
+  Just like the name implies, OOP uses objects as the focal point in the manipulation of logic. The core pillars of OOP include 
+  encapsulation (encapsulating or wrapping all data, so it can only be accessed within its own class.) , abstraction (ensuring of necessary data/information is exposed), inheritance (essentially like creating an instance with objects 
+  or passing props from parent to child), polymorphism (multiple ways of data manipulation to attain the same end result)
 
-
+****  Reference ****
+https://www.freecodecamp.org/news/object-oriented-programming-concepts-21bb035f7260/
 
 4. What is the difference between a Float and an Integer in Ruby?
 
   Your answer:
+  In Ruby an integer is a whole number while a float is a fractional number or a decimal number. 
 
   Researched answer:
+  Floats and integers fall under the parent class Numeric. Ihe integer class contains two subclasses fixnum and bignum (larger whole numbers).
+  Floats are also known as imprecise decimal numbers, while the BigDecimal class is used for precise decimal numbers.
```

Comment:
Thanks for that info. A definite separation between floats and integers is the aspect of a decimal on floats.